NDTensors/src/lib/GradedAxes/src/fusion.jl

```diff
@@ -83,24 +105,38 @@ end
 # Get the permutation for sorting, then group by common elements.
 # groupsortperm([2, 1, 2, 3]) == [[2], [1, 3], [4]]
 function blockmergesortperm(a::AbstractBlockedUnitRange)
-  # If it is dual, reverse the sorting so the sectors
-  # end up sorted in the same way whether or not the space
-  # is dual.
-  # TODO: Figure out how to deal with dual sectors.
-  # TODO: `rev=isdual(a)`  may not be correct for symmetries beyond `U(1)`.
-  ## return Block.(groupsortperm(nondual_sectors(a); rev=isdual(a)))
   return Block.(groupsortperm(blocklabels(a)))
 end
 
 # Used by `TensorAlgebra.splitdims` in `BlockSparseArraysGradedAxesExt`.
 invblockperm(a::Vector{<:Block{1}}) = Block.(invperm(Int.(a)))
 
-# Used by `TensorAlgebra.fusedims` in `BlockSparseArraysGradedAxesExt`.
-function blockmergesortperm(a::GradedUnitRange)
-  # If it is dual, reverse the sorting so the sectors
-  # end up sorted in the same way whether or not the space
-  # is dual.
-  # TODO: Figure out how to deal with dual sectors.
-  # TODO: `rev=isdual(a)`  may not be correct for symmetries beyond `U(1)`.
-  return Block.(groupsortperm(blocklabels(a)))
+function blockmergesortperm(a::UnitRangeDual)
+  return Block.(groupsortperm(blocklabels(nondual(a))))
+end
+
+function blockmergesort(g::AbstractGradedUnitRange)
+  glabels = blocklabels(g)
+  gblocklengths = blocklengths(g)
+  new_blocklengths = map(
+    la -> labelled(sum(gblocklengths[findall(==(la), glabels)]; init=0), la),
+    sort(unique(glabels)),
+  )
+  return GradedAxes.gradedrange(new_blocklengths)
+end
+
+blockmergesort(g::UnitRangeDual) = dual(blockmergesort(flip(g)))
+blockmergesort(g::OneToOne) = g
```

Comment:
```suggestion
blockmergesort(g::AbstractUnitRange) = g
```
Again, it seems a bit strange having special definitions for `OneToOne`, maybe one way to think about it is defining trivial fallbacks that work for any non-blocked ranges.

Reply:
I plan to deal with this in #1531

Reply:
How will it be addressed in #1531?

Reply:
I plan to systematically check for different types combination. If you prefer I can make the change here.